language/evm/move-to-yul/src/functions.rs

```diff
@@ -107,48 +108,260 @@ impl<'a> FunctionGenerator<'a> {
+                            );
+                        }
+                    });
+                }
+                emit!(ctx.writer, "default ");
+                if let StacklessSCG::BasicBlock {
+                    start_offset,
+                    end_offset,
+                } = **if_true
+                {
+                    ctx.emit_block(|| {
+                        for offs in start_offset..end_offset + 1 {
+                            self.bytecode(
+                                ctx,
+                                fun_id,
+                                target,
+                                &label_map,
+                                &code[offs as usize],
+                                false,
+                                true,
+                            );
                         }
+                    });
+                }
+            }
+            // TODO: need to emit codes of loops based on StacklessSCG::LoopBlock
+            // based on new bytecodes of Break and Continue
+            StacklessSCG::LoopBlock {
+                loop_header,
+                loop_body,
+            } => {
+                self.emit_scg(&*loop_header, &cfg, code, ctx, fun_id, target);
+                emitln!(ctx.writer, "for {} true {} ");
+                ctx.emit_block(|| {
+                    for loop_body_scg in loop_body {
+                        self.emit_scg(loop_body_scg, &cfg, code, ctx, fun_id, target);
                     }
-                })
+                });
             }
-        });
-        emitln!(ctx.writer)
+        }
+    }
+
+    pub fn push_non_loop_scg(
+        scg_vec: &mut Vec<StacklessSCG>,
+        blk_id: BlockId,
+        cfg: &StacklessControlFlowGraph,
+        visited_blocks: &mut BTreeSet<BlockId>,
+        code: &[Bytecode],
+    ) {
+        let label_map = Self::compute_label_map(cfg, code);
+        let get_block = |l| label_map.get(l).expect("label has corresponding block");
+        if let BlockContent::Basic { lower, upper } = cfg.content(blk_id) {
+            let mut start = *lower;
+            for offs in *lower..*upper + 1 {
+                match &code[offs as usize] {
+                    Bytecode::Branch(_, if_t, if_f, cond) => {
+                        scg_vec.push(StacklessSCG::BasicBlock {
+                            start_offset: start as usize,
+                            end_offset: offs as usize,
+                        });
+                        start = offs;
+
+                        let if_block_id = get_block(if_t);
+                        let else_block_id = get_block(if_f);
+                        let if_else_scg = StacklessSCG::IfBlock {
+                            cond: *cond,
+                            if_true: Box::new(StacklessSCG::new(*if_block_id, &cfg)),
+                            if_false: Box::new(StacklessSCG::new(*else_block_id, &cfg)),
+                        };
+                        visited_blocks.insert(*if_block_id);
```

Comment:
I'm not sure whether this is correct. You have not fully processed if/else blocks, which can contain themselves ifs and loops. Marking them as done seems to ignore them for further processing, AFAICT. This algorithm, I guess, needs to be somehow recursive.

Reply:
IIUC, the insertion is to avoid pushing duplicates to scg_vec.
Re nested if-else, I followed your comment and counted on the processing of the scg_vec to recursively emit the inner if-else.